### ISSUE TITLE
rancher cli apps ls broken

### ISSUE THREAD

**Issue by benzman81** [rancher/rancher#23963]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
calling "rancher cli apps ls"

**Result:**
```
ID        NAME      STATE     CATALOG   TEMPLATE   VERSION
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x822d19]

goroutine 1 [running]:
github.com/rancher/cli/cmd.appLs(0xc0000b49a0, 0x0, 0x0)
        /go/src/github.com/rancher/cli/cmd/app.go:312 +0x749
github.com/rancher/cli/vendor/github.com/urfave/cli.HandleAction(0x8e3ce0, 0x9f2010, 0xc0000b49a0, 0x0, 0xc00012c720)
        /go/src/github.com/rancher/cli/vendor/github.com/urfave/cli/app.go:501 +0xc8
github.com/rancher/cli/vendor/github.com/urfave/cli.Command.Run(0x9cf6ac, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0, 0x9d1194, 0x9, 0x0, ...)
        /go/src/github.com/rancher/cli/vendor/github.com/urfave/cli/command.go:165 +0x487
github.com/rancher/cli/vendor/github.com/urfave/cli.(*App).RunAsSubcommand(0xc0000fa380, 0xc0000b46e0, 0x0, 0x0)
        /go/src/github.com/rancher/cli/vendor/github.com/urfave/cli/app.go:383 +0x850
github.com/rancher/cli/vendor/github.com/urfave/cli.Command.startApp(0x9cfab2, 0x4, 0x0, 0x0, 0xc0000291a0, 0x1, 0x1, 0x9eb948, 0x74, 0x0, ...)
        /go/src/github.com/rancher/cli/vendor/github.com/urfave/cli/command.go:330 +0x821
github.com/rancher/cli/vendor/github.com/urfave/cli.Command.Run(0x9cfab2, 0x4, 0x0, 0x0, 0xc0000291a0, 0x1, 0x1, 0x9eb948, 0x74, 0x0, ...)
        /go/src/github.com/rancher/cli/vendor/github.com/urfave/cli/command.go:103 +0x809
github.com/rancher/cli/vendor/github.com/urfave/cli.(*App).Run(0xc0000fa1c0, 0xc000020980, 0x3, 0x4, 0x0, 0x0)
        /go/src/github.com/rancher/cli/vendor/github.com/urfave/cli/app.go:259 +0x6e3
main.mainErr(0xc000066058, 0x0)
        /go/src/github.com/rancher/cli/main.go:117 +0x18bb
main.main()
        /go/src/github.com/rancher/cli/main.go:57 +0x22
```

**Other details that may be helpful:**
I have only one app installed (via rancher app install for a local helm file [not in catalog]=

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.3
- Installation option (single install/HA): single install
- Cli verion: 2.3.1 and 2.2.0 tested

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported):
- Machine type (cloud/VM/metal) and specifications (CPU/memory):
- Kubernetes version (use `kubectl version`):

```
(paste the output here)
```

- Docker version (use `docker version`):

```
Client: Docker Engine - Community
 Version:           19.03.1
 API version:       1.40
 Go version:        go1.12.5
 Git commit:        74b1e89
 Built:             Thu Jul 25 21:21:05 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.1
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.5
  Git commit:       74b1e89
  Built:            Thu Jul 25 21:19:41 2019
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.2.6
  GitCommit:        894b81a4b802e4eb2a91d1ce216b8817763c29fb
 runc:
  Version:          1.0.0-rc8
  GitCommit:        425e105d5a03fabd737a126ad93d62a9eeede87f
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683

```

**Comment by superseb** [comment]
Can you share steps to reproduce this? The source used to create the app and the exact CLI command so we can use that to verify the fix?

**Comment by dramich** [comment]
To test:

1. Launch app, I used docker-registry
2. Edit app using kubectl to delete externalID field
3. Edit apprevision using kubectl to delete Spec
4. Enjoy the panic

**Comment by jiaqiluo** [comment]
The bug is reproduced with Rancher CLI `v2.3.1`
Results:
```
➜  rancher-v2.3.1 ./rancher apps ls
ID        NAME      STATE     CATALOG   TEMPLATE   VERSION
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1426489]

goroutine 1 [running]:
github.com/rancher/cli/cmd.appLs(0xc0000f66e0, 0x0, 0x0)
	/go/src/github.com/rancher/cli/cmd/app.go:312 +0x749
github.com/rancher/cli/vendor/github.com/urfave/cli.HandleAction(0x14e7260, 0x15f5bb0, 0xc0000f66e0, 0x0, 0xc0001cc180)
	/go/src/github.com/rancher/cli/vendor/github.com/urfave/cli/app.go:501 +0xc8
github.com/rancher/cli/vendor/github.com/urfave/cli.Command.Run(0x15d357a, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0, 0x15d5044, 0x9, 0x0, ...)
	/go/src/github.com/rancher/cli/vendor/github.com/urfave/cli/command.go:165 +0x487
github.com/rancher/cli/vendor/github.com/urfave/cli.(*App).RunAsSubcommand(0xc0001401c0, 0xc0000f6580, 0x0, 0x0)
	/go/src/github.com/rancher/cli/vendor/github.com/urfave/cli/app.go:383 +0x850
github.com/rancher/cli/vendor/github.com/urfave/cli.Command.startApp(0x15d3983, 0x4, 0x0, 0x0, 0xc000180210, 0x1, 0x1, 0x15ef4e5, 0x74, 0x0, ...)
	/go/src/github.com/rancher/cli/vendor/github.com/urfave/cli/command.go:330 +0x821
github.com/rancher/cli/vendor/github.com/urfave/cli.Command.Run(0x15d3983, 0x4, 0x0, 0x0, 0xc000180210, 0x1, 0x1, 0x15ef4e5, 0x74, 0x0, ...)
	/go/src/github.com/rancher/cli/vendor/github.com/urfave/cli/command.go:103 +0x809
github.com/rancher/cli/vendor/github.com/urfave/cli.(*App).Run(0xc0001aa000, 0xc0001740c0, 0x3, 0x4, 0x0, 0x0)
	/go/src/github.com/rancher/cli/vendor/github.com/urfave/cli/app.go:259 +0x6e3
main.mainErr(0xc00002e118, 0x0)
	/go/src/github.com/rancher/cli/main.go:117 +0x18bb
main.main()
	/go/src/github.com/rancher/cli/main.go:57 +0x22
```

**Comment by jiaqiluo** [comment]
The bug fix is validated with Rancher CLI `v2.3.2` https://github.com/rancher/cli/releases/tag/v2.3.2

Results:
instead of the panic, now the output is an error message 

```
➜  rancher-v2.3.2 ./rancher apps
ID        NAME      STATE     CATALOG   TEMPLATE   VERSION
FATA[0000] can't parse info from app
➜  rancher-v2.3.2 ./rancher --version
rancher version v2.3.2
```

**Comment by benzman81** [comment]
Works like a charm with v2.3.2.
